contracts/proxy/proxy.cpp

```diff
@@ -96,13 +94,10 @@ extern "C" {
        if ( code == N(eosio)) {
           if (action == N(onerror)) {
              apply_onerror(receiver, deferred_transaction::from_current_action());
-          } if( action == N(transfer) ) {
-             // Comment this out for now so that the contract compiles, this will change with refactoring to use eosio.token
-             //        apply_transfer(receiver, code, unpack_action_data<eosiosystem::contract<N(eosio.system)>::currency::transfer_memo>());
           }
-       } else if ( code == N(currency) ) {
+       } else if ( code == N(eosio.token) ) {
           if( action == N(transfer) ) {
-             apply_transfer(receiver, code, unpack_action_data<eosio::currency::transfer>());
+             apply_transfer(receiver, code, unpack_action_data<transfer_args>());
```

Comment:
Why does this create its own version of an action structure that is owned by the eosio.token contract?